libraries/chain/controller.cpp

```diff
@@ -3096,22 +3133,13 @@ void controller::set_action_blacklist( const flat_set< pair<account_name, action
 void controller::set_key_blacklist( const flat_set<public_key_type>& new_key_blacklist ) {
    my->conf.key_blacklist = new_key_blacklist;
 }
-void controller::set_s_header( const s_header& s_header ) {
-   //  ilog("CONTROLLER - setting s header: ${s}", ("s", s_header.to_string()));
-   SYS_ASSERT( my->pending, block_validate_exception, "it is not valid to set_s_header when there is no pending block");
-   SYS_ASSERT( std::holds_alternative<building_block>(my->pending->_block_stage), block_validate_exception, "already called finalize_block");
-   // SYS_ASSERT( std::get<assembled_block>(pending->_block_stage)._id == id, block_validate_exception, "mismatching block id" );
+
+void controller_impl::set_s_header( deque<s_header>&& s_headers ) {
```

Comment:
any reason why this method name is singular instead of plural?

Reply:
Nope. Will change.